message_edit: Show modal if user cannot resolve topics.

sahil839 opened the pull request:
We show a modal if user is not allowed to resolve or unresolve topics due to time limit.
The modal just contains the text mentioning user cannot resolve topic without stating the count
of messages that are withing the time limit as we do not recommend partial resolving of topics.

<!-- Describe your pull request here.-->
 <!-- Issue link, or clear description.-->

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**
![resolve-unresolve-topic](https://user-images.githubusercontent.com/35494118/231414912-2daccc1e-75c9-4f17-943e-ccb8a26e1709.gif)


![Screenshot from 2023-04-12 14-53-51](https://user-images.githubusercontent.com/35494118/231414962-66b8e49c-4507-41f7-b1ac-a68f1c063b30.png)

![Screenshot from 2023-04-12 14-54-25](https://user-images.githubusercontent.com/35494118/231415021-05d7ba67-c860-4001-99ae-8c3ca4657ed2.png)



<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [ ] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

sahil839 commented:
Unresolving the topic from warning banner in compose box -
![resolve-topic-warning](https://user-images.githubusercontent.com/35494118/231484673-c618ae88-3dc8-4152-a498-c689cae52fa6.gif)

timabbott commented:
The implementation lgtm aside from polishing the modal heading/content, which I'll leave for @alya to review.

alya commented:
Updated wording discussed on CZO:

> You do not have permission to resolve topics with messages older than X in this organization. 

(where X is relative times, based on what the org setting is ("2 hours", "1 week", etc.))

sahil839 commented:
Updated the PR -
- Changed the text in the modal as discussed.
- We now do not show separate modal when resolving or unresolving topic using "Move topic"/"Move message" modal and consider it as a simple message move.

![Screenshot from 2023-04-14 08-36-34](https://user-images.githubusercontent.com/35494118/231933436-e4a64a2e-b312-4967-9904-489381f6a681.png)
![Screenshot from 2023-04-14 08-37-36](https://user-images.githubusercontent.com/35494118/231933464-5415e67e-77ee-48a6-88e2-81cf470cd719.png)

timabbott commented on web/src/timerender.ts:
```diff
@@ -455,3 +455,27 @@ export function get_full_datetime(time: Date): string {
+        return time_limit_in_days.toString() + " day";
+    }
+    return time_limit_in_days.toString() + " days";
+}
```
This doesn't work from an i18n standpoint -- we're now combining English strings with numbers.

You should be able to fix it by just passing the time values into a translated string as the variable for the number, with ICU Plurals for the day/days part.

timabbott commented on web/src/message_edit.js:
```diff
@@ -603,9 +605,94 @@ export function start($row, edit_box_open_callback) {
+        },
+        {N: time_limit},
+    );
+}
```
This looks correct but indeed a bit verbose. @alya I think the main optimization we could do is deciding we don't need separate strings for "unresolve topics".

timabbott replied:
Checked with Alya and she thinks it's worth it, so integrating.

timabbott commented:
Merged, thanks @sahil839!
